## 71% of end-end tests have passed
```
Total executed tests: 7
Failed tests: 2
Expected to fail tests: 0
Passed tests: 5
```
```
IDs of failed tests: 727230,702843 
```
<h3>Failed tests (2)</h3><details><summary>Click to expand</summary><br/><li><a href="https://ci.status.im/job/status-mobile/job/e2e/job/status-app-prs-rerun/parambuild/?BRANCH_NAME=feat/wc-deeplink&PR_ID=21050&APK_NAME=21050.apk&TR_CASE_IDS=727230,702843">Rerun failed tests</a></li><br/><h4>Class TestWalletMultipleDevice:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3656992">test_wallet_send_asset_from_drawer</a>, id: 727230 </b></td></tr><tr><td><p><blockquote><div>Device 2: `Text` is `0.04459 ETH` </div><div>Device 1: `Text` is `0.44452 ETH` </div></blockquote></p>

```
critical/test_wallet.py:190: in test_wallet_send_asset_from_drawer
    self.errors.verify_no_errors()
base_test_case.py:191: in verify_no_errors
    pytest.fail('\n '.join([self.errors.pop(0) for _ in range(len(self.errors))]))
 Sender balance is not updated on Etherscan, it is 0.4445 but expected to be 0.4446
```

<br/><br/></td></tr></tbody></table><h4>Class TestCommunityMultipleDeviceMerged:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3656989">test_community_message_edit</a>, id: 702843 </b></td></tr><tr><td><p><blockquote><div>Device 1: Find `Text` by `xpath`: `//android.view.ViewGroup[@content-desc='chat-item']//android.widget.TextView[contains(@text,'Message BEFORE edit')]` </div><div>Device 1: Long press on `Text` until expected element is shown </div></blockquote></p>

```
critical/chats/test_public_chat_browsing.py:373: in test_community_message_edit
    self.channel_1.edit_message_in_chat(message_before_edit, message_after_edit)
../views/chat_view.py:1015: in edit_message_in_chat
    self.chat_element_by_text(message_to_edit).message_body.long_press_until_element_is_shown(element)
../views/base_element.py:331: in long_press_until_element_is_shown
    action.long_press(element).release().perform()
../../../../status-app-prs-rerun@tmp/venv/lib/python3.10/site-packages/appium/webdriver/common/touch_action.py:174: in perform
    self._driver.execute(Command.TOUCH_ACTION, params)
../../../../status-app-prs-rerun@tmp/venv/lib/python3.10/site-packages/selenium/webdriver/remote/webdriver.py:345: in execute
    self.error_handler.check_response(response)
../../../../status-app-prs-rerun@tmp/venv/lib/python3.10/site-packages/appium/webdriver/errorhandler.py:122: in check_response
    raise exception_class(msg=message, stacktrace=format_stacktrace(stacktrace))
 The element 'By.xpath: //android.view.ViewGroup[@content-desc='chat-item']//android.widget.TextView[contains(@text,'Message BEFORE edit')]' is not linked to the same object in DOM anymore; For documentation on this error, please visit: https://www.selenium.dev/documentation/webdriver/troubleshooting/errors#stale-element-reference-exception
E   Stacktrace:
E   io.appium.uiautomator2.common.exceptions.StaleElementReferenceException: The element 'By.xpath: //android.view.ViewGroup[@content-desc='chat-item']//android.widget.TextView[contains(@text,'Message BEFORE edit')]' is not linked to the same object in DOM anymore
E   	at io.appium.uiautomator2.model.ElementsCache.restore(ElementsCache.java:122)
E   	at io.appium.uiautomator2.model.ElementsCache.get(ElementsCache.java:153)
E   	at io.appium.uiautomator2.handler.Location.safeHandle(Location.java:23)
E   	at io.appium.uiautomator2.handler.request.SafeRequestHandler.handle(SafeRequestHandler.java:59)
E   	at io.appium.uiautomator2.server.AppiumServlet.handleRequest(AppiumServlet.java:277)
E   	at io.appium.uiautomator2.server.AppiumServlet.handleHttpRequest(AppiumServlet.java:271)
E   	at io.appium.uiautomator2.http.ServerHandler.channelRead(ServerHandler.java:68)
E   	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:366)
E   	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:352)
E   	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:345)
E   	at io.netty.handler.codec.MessageToMessageDecoder.channelRead(MessageToMessageDecoder.java:102)
E   	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:366)
E   	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:352)
E   	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:345)
E   	at io.netty.channel.CombinedChannelDuplexHandler$DelegatingChannelHandlerContext.fireChannelRead(CombinedChannelDuplexHandler.java:435)
E   	at io.netty.handler.codec.ByteToMessageDecoder.fireChannelRead(ByteToMessageDecoder.java:293)
E   	at io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:267)
E   	at io.netty.channel.CombinedChannelDuplexHandler.channelRead(CombinedChannelDuplexHandler.java:250)
E   	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:366)
E   	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:352)
E   	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:345)
E   	at io.netty.handler.timeout.IdleStateHandler.channelRead(IdleStateHandler.java:266)
E   	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:366)
E   	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:352)
E   	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:345)
E   	at io.netty.channel.DefaultChannelPipeline$HeadContext.channelRead(DefaultChannelPipeline.java:1294)
E   	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:366)
E   	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:352)
E   	at io.netty.channel.DefaultChannelPipeline.fireChannelRead(DefaultChannelPipeline.java:911)
E   	at io.netty.channel.nio.AbstractNioByteChannel$NioByteUnsafe.read(AbstractNioByteChannel.java:131)
E   	at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:611)
E   	at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:552)
E   	at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:466)
E   	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:438)
E   	at io.netty.util.concurrent.SingleThreadEventExecutor$2.run(SingleThreadEventExecutor.java:140)
E   	at io.netty.util.concurrent.DefaultThreadFactory$DefaultRunnableDecorator.run(DefaultThreadFactory.java:144)
E   	at java.lang.Thread.run(Thread.java:1012)
```

<br/><br/><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/26004e5bef9541a18c05aa2d6ffb44b5?auth=dd9ccf5f0a9c41ddeb1457267dcb58e4#160">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/592482c0959f41508798f19ded045c90?auth=ce3e85b93a2b6e3303fbd3e007b0ea54#107">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table></details><h3>Passed tests (5)</h3><details><summary>Click to expand</summary><br/><br/><h4>Class TestWalletMultipleDevice:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3656991">test_wallet_send_eth</a>, id: 727229 </b></td></tr><tr><td></td></tr></tbody></table><h4>Class TestCommunityOneDeviceMerged:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3656987">test_community_copy_and_paste_message_in_chat_input</a>, id: 702742 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/dbb05e0bfb1e44468487800f31993246?auth=40285483a4f2661febf3bd72a3e23d79#53">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3656990">test_restore_multiaccount_with_waku_backup_remove_switch</a>, id: 703133 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/dbb05e0bfb1e44468487800f31993246?auth=40285483a4f2661febf3bd72a3e23d79#82">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestOneToOneChatMultipleSharedDevicesNewUi:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3656988">test_1_1_chat_non_latin_messages_stack_update_profile_photo</a>, id: 702745 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/a9e05220a6b64f6bb335df5ed48ee028?auth=932056a3fed2ae9ff684be54d4a325bb#101">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/2cd4d63e4c2e40a0ad34731f694ea31e?auth=e1d9a094de78fed746c9d1e0bd6aa1ca#90">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestWalletOneDevice:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3656993">test_wallet_add_remove_regular_account</a>, id: 727231 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/b7de1edabeda4ac9ba505e7bf286e183?auth=759cef2861ab71e656fbfab3fd5648bd#39">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table></details>

--- a/src/status_im/common/router.cljs
+++ b/src/status_im/common/router.cljs
@@ -60,7 +60,9 @@
                      ["wallet/" :account]                       :wallet-account
                      [user-with-data-path :user-data]           :user
                      "c"                                        :community
-                     "u"                                        :user}
+                     "u"                                        :user
+                     #"wc:([^\?]+)"                             :wallet-connect
+                     "wc"                                       :wallet-connect}
     ethereum-scheme eip-extractor}])
 
 (defn parse-query-params
@@ -75,6 +77,11 @@
     (when-not (string/blank? fragment)
       fragment)))
 
+(defn remove-scheme
+  [uri]
+  (let [[_ url] (string/split uri #"://")]
+    url))
+
 (defn match-uri
   [uri]
   (let [;; bidi has trouble parse path with `=` in it extract `=` here and add back to parsed
@@ -88,13 +95,17 @@
         uri-without-equal-in-path
         (if equal-end-of-base64url (string/replace-first uri equal-end-of-base64url "") uri)
 
+        ;;bidi has issue to detect wc regex with /?, couldn't fine any other workaround
+        normalize-url
+        (if (string/includes? uri "//wc:") (string/replace-first uri "/?" "?") uri-without-equal-in-path)
+
         ;; fragment is the one after `#`, usually user-id, ens-name, community-id
         fragment (parse-fragment uri)
         ens? (utils.ens/is-valid-eth-name? fragment)
         compressed-key? (validators/valid-compressed-key? fragment)
 
-        {:keys [handler route-params] :as parsed}
-        (assoc (bidi/match-route routes uri-without-equal-in-path)
+        {:keys [handler route-params query-params] :as parsed}
+        (assoc (bidi/match-route routes normalize-url)
                :uri          uri
                :query-params (parse-query-params uri))]
     (cond-> parsed
@@ -131,7 +142,13 @@
       (update-in [:route-params :user-data] #(str % equal-end-of-base64url))
 
       (and (= handler :user) compressed-key?)
-      (assoc-in [:route-params :user-id] fragment))))
+      (assoc-in [:route-params :user-id] fragment)
+
+      (= handler :wallet-connect)
+      (assoc-in [:route-params :uri]
+       (if (get query-params "uri")
+         (get query-params "uri")
+         (remove-scheme normalize-url))))))
 
 (defn match-contact-async
   [chain {:keys [user-id ens-name]} callback]
@@ -327,6 +344,10 @@
       ;; NOTE: removed in `match-uri`, might need this in the future
       (= handler :wallet-account)
       (cb (match-wallet-account route-params))
+
+      (= handler :wallet-connect)
+      (cb {:type :wallet-connect
+           :uri  (:uri route-params)})
 
       (address/address? uri)
       (cb (address->eip681 uri))

--- a/src/status_im/common/router.cljs
+++ b/src/status_im/common/router.cljs
@@ -38,8 +38,11 @@
   [path]
   (map #(str % path) status-web-urls))
 
-
 (def handled-schemes (set (into uri-schemes status-web-urls)))
+
+(defn wc-normalize-uri
+  [uri]
+  (if (string/includes? uri "//wc:") (string/replace-first uri "/?" "?") uri))
 
 (def group-chat-extractor
   {[#"(.*)" :params] {""  :group-chat
@@ -96,8 +99,8 @@
         (if equal-end-of-base64url (string/replace-first uri equal-end-of-base64url "") uri)
 
         ;;bidi has issue to detect wc regex with /?, couldn't fine any other workaround
-        normalize-url
-        (if (string/includes? uri "//wc:") (string/replace-first uri "/?" "?") uri-without-equal-in-path)
+        normalize-uri
+        (wc-normalize-uri uri-without-equal-in-path)
 
         ;; fragment is the one after `#`, usually user-id, ens-name, community-id
         fragment (parse-fragment uri)
@@ -105,7 +108,7 @@
         compressed-key? (validators/valid-compressed-key? fragment)
 
         {:keys [handler route-params query-params] :as parsed}
-        (assoc (bidi/match-route routes normalize-url)
+        (assoc (bidi/match-route routes normalize-uri)
                :uri          uri
                :query-params (parse-query-params uri))]
     (cond-> parsed
@@ -148,7 +151,7 @@
       (assoc-in [:route-params :uri]
        (if (get query-params "uri")
          (get query-params "uri")
-         (remove-scheme normalize-url))))))
+         (remove-scheme normalize-uri))))))
 
 (defn match-contact-async
   [chain {:keys [user-id ens-name]} callback]

--- a/src/status_im/common/universal_links.cljs
+++ b/src/status_im/common/universal_links.cljs
@@ -105,6 +105,11 @@
   [full-url]
   (log/info "universal-links: no handler for " full-url))
 
+(rf/defn handle-wallet-connect
+  [_ uri]
+  (log/info "universal-links: handle-wallet-connect" uri)
+  {:dispatch [:wallet-connect/on-scan-connection uri]})
+
 (defn dispatch-url
   "Dispatch url so we can get access to re-frame/db"
   [url]
@@ -117,7 +122,7 @@
 
 (rf/defn on-handle
   {:events [:universal-links/match-value]}
-  [cofx url {:keys [type chat-id community-id] :as data}]
+  [cofx url {:keys [type chat-id community-id uri] :as data}]
   (case type
     :group-chat         (handle-group-chat cofx data)
     :private-chat       (handle-private-chat cofx data)
@@ -130,6 +135,7 @@
     :browser            (handle-browse cofx data)
     :eip681             (handle-eip681 cofx data)
     :wallet-account     (handle-wallet-account cofx data)
+    :wallet-connect     (handle-wallet-connect cofx uri)
     (handle-not-found url)))
 
 (rf/defn route-url

--- a/src/status_im/contexts/wallet/wallet_connect/core.cljs
+++ b/src/status_im/contexts/wallet/wallet_connect/core.cljs
@@ -59,12 +59,11 @@
          (filter (fn [session]
                    (= (utils.string/remove-trailing-slash dapp-url)
                       (utils.string/remove-trailing-slash (get session :url)))))
-         (first))))
+         first)))
 
 (defn get-dapp-redirect-url
   [session]
-  (let [peer-metadata (get-in session [:peer :metadata])]
-    (get-in peer-metadata [:redirect :native])))
+  (get-in session [:peer :metadata :redirect :native]))
 
 (defn get-db-current-request-params
   [db]

--- a/src/status_im/contexts/wallet/wallet_connect/core_test.cljs
+++ b/src/status_im/contexts/wallet/wallet_connect/core_test.cljs
@@ -1,0 +1,27 @@
+(ns status-im.contexts.wallet.wallet-connect.core-test
+  (:require
+    [cljs.test :refer-macros [deftest is testing]]
+    [status-im.contexts.wallet.wallet-connect.core :as sut]))
+
+(deftest get-current-request-dapp-test
+  (testing "returns the correct dapp based on the request's origin"
+    (let [request  {:event {:verifyContext {:verified {:origin "https://dapp.com"}}}}
+          sessions [{:url "https://dapp.com"}
+                    {:url "https://anotherdapp.com"}]]
+      (is (= {:url "https://dapp.com"}
+             (sut/get-current-request-dapp request sessions)))))
+
+  (testing "returns nil if no matching dapp is found"
+    (let [request  {:event {:verifyContext {:verified {:origin "https://dapp.com"}}}}
+          sessions [{:url "https://anotherdapp.com"}]]
+      (is (nil? (sut/get-current-request-dapp request sessions))))))
+
+(deftest get-dapp-redirect-url-test
+  (testing "returns the native redirect URL if it exists"
+    (let [session {:peer {:metadata {:redirect {:native "native://redirect-url"}}}}]
+      (is (= "native://redirect-url"
+             (sut/get-dapp-redirect-url session)))))
+
+  (testing "returns nil if no redirect URL is found"
+    (let [session {:peer {:metadata {}}}]
+      (is (nil? (sut/get-dapp-redirect-url session))))))

--- a/src/status_im/contexts/wallet/wallet_connect/events.cljs
+++ b/src/status_im/contexts/wallet/wallet_connect/events.cljs
@@ -317,13 +317,17 @@
 (rf/reg-event-fx
  :wallet-connect/persist-session
  (fn [_ [session-info]]
-   {:fx [[:json-rpc/call
-          [{:method     "wallet_addWalletConnectSession"
-            :params     [(js/JSON.stringify session-info)]
-            :on-success (fn []
-                          (log/info "Wallet Connect session persisted")
-                          (rf/dispatch [:wallet-connect/fetch-persisted-sessions]))
-            :on-error   #(log/info "Wallet Connect session persistence failed" %)}]]]}))
+   (let [redirect-url (-> session-info
+                          (js->clj :keywordize-keys true)
+                          (wallet-connect-core/get-dapp-redirect-url))]
+     {:fx [[:json-rpc/call
+            [{:method     "wallet_addWalletConnectSession"
+              :params     [(js/JSON.stringify session-info)]
+              :on-success (fn []
+                            (log/info "Wallet Connect session persisted")
+                            (rf/dispatch [:wallet-connect/fetch-persisted-sessions])
+                            (rf/dispatch [:wallet-connect/redirect-to-dapp redirect-url]))
+              :on-error   #(log/info "Wallet Connect session persistence failed" %)}]]]})))
 
 (rf/reg-event-fx
  :wallet-connect/disconnect-session

--- a/src/status_im/contexts/wallet/wallet_connect/responding_events.cljs
+++ b/src/status_im/contexts/wallet/wallet_connect/responding_events.cljs
@@ -1,6 +1,5 @@
 (ns status-im.contexts.wallet.wallet-connect.responding-events
   (:require [re-frame.core :as rf]
-            [react-native.platform :as platform]
             [react-native.wallet-connect :as wallet-connect]
             [status-im.constants :as constants]
             [status-im.contexts.wallet.wallet-connect.core :as wallet-connect-core]
@@ -134,18 +133,19 @@
                                           :event                :wallet-connect/send-response
                                           :wallet-connect-event event}))
                :on-success  (fn []
-                              (when platform/android?
-                                (rf/dispatch [:wallet-connect/redirect-to-dapp]))
+                              (rf/dispatch [:wallet-connect/redirect-to-dapp])
                               (log/info "Successfully sent Wallet Connect response to dApp"))}]]}))))
 
 (rf/reg-event-fx
  :wallet-connect/redirect-to-dapp
- (fn [{:keys [db]} _]
-   (let [current-request (get db :wallet-connect/current-request)
-         sessions        (get db :wallet-connect/sessions)
-         dapp            (wallet-connect-core/get-current-request-dapp current-request sessions)
-         session         (transforms/json->clj (dapp :sessionJson))
-         redirect-url    (wallet-connect-core/get-dapp-redirect-url session)]
+ (fn [{:keys [db]} [url]]
+   (let [redirect-url (or url
+                          (->> (get db :wallet-connect/current-request)
+                               (wallet-connect-core/get-current-request-dapp
+                                (get db :wallet-connect/sessions))
+                               :sessionJson
+                               transforms/json->clj
+                               wallet-connect-core/get-dapp-redirect-url))]
      {:fx [[:open-url redirect-url]]})))
 
 (rf/reg-event-fx

--- a/src/status_im/contexts/wallet/wallet_connect/responding_events_test.cljs
+++ b/src/status_im/contexts/wallet/wallet_connect/responding_events_test.cljs
@@ -1,0 +1,27 @@
+(ns status-im.contexts.wallet.wallet-connect.responding-events-test
+  (:require
+    [cljs.test :refer-macros [is]]
+    matcher-combinators.test
+    [re-frame.db :as rf-db]
+    [status-im.contexts.wallet.wallet-connect.core :as wallet-connect-core]
+    status-im.contexts.wallet.wallet-connect.responding-events
+    [test-helpers.unit :as h]
+    [utils.transforms :as transforms]))
+
+(h/deftest-event :wallet-connect/redirect-to-dapp
+  [event-id dispatch]
+  (let [current-request {:event {:verifyContext {:verified {:origin "https://dapp.com"}}}}
+        session-json    "{\"peer\":{\"metadata\":{\"redirect\":{\"native\":\"native://redirect-url\"}}}}"
+        sessions        [{:url         "https://dapp.com"
+                          :sessionJson session-json}]]
+    (reset! rf-db/app-db {:wallet-connect {:current-request current-request
+                                           :sessions        sessions}})
+    (with-redefs [wallet-connect-core/get-current-request-dapp
+                  (fn [_ _] (first sessions))
+                  transforms/json->clj
+                  (fn [json] (js/JSON.parse json))
+                  wallet-connect-core/get-dapp-redirect-url
+                  (fn [_] "native://redirect-url")]
+
+      (is (match? {:fx [[:open-url "native://redirect-url"]]}
+                  (dispatch [event-id]))))))

